libs/vm/src/analyser.cpp

```diff
@@ -494,9 +499,12 @@ void Analyser::ValidateFunctionAnnotations(NodePtr const &function_node)
   for (auto const &annotation : annotations_node->children)
   {
     auto const &text = annotation->text;
-    if (text != INIT_ANNOTATION && text != ACTION_ANNOTATION && text != QUERY_ANNOTATION &&
-        text != PROBLEM_ANNOTATION && text != OBJECTIVE_ANNOTATION && text != WORK_ANNOTATION &&
-        text != CLEAR_ANNOTATION)
+    bool        test_annotation =
+        (!allow_test_annotations_) || (text != TEST_ANNOTATION && text != TEST_INIT_ANNOTATION);
+
+    if (test_annotation && text != INIT_ANNOTATION && text != ACTION_ANNOTATION &&
+        text != QUERY_ANNOTATION && text != PROBLEM_ANNOTATION && text != OBJECTIVE_ANNOTATION &&
+        text != WORK_ANNOTATION && text != CLEAR_ANNOTATION)
```

Comment:
One thing we could improve this year is to have more meaningful error messages - I think we can do better here than "Invalid annotation".

```
if (text == TEST_ANNOTATION || text == TEST_INIT_ANNOTATION)
{
  if (!allow_test_annotations_)
  {
    AddError(annotation->line, "Test annotations are disabled '" + text + "'");
  }

  return;
}

if (text != INIT_ANNOTATION && text != ACTION_ANNOTATION && ...
```